tiatoolbox/dataloader/wsireader.py

```diff
@@ -836,18 +836,42 @@ def info(self):
             WSIMeta: containing meta information.
 
         """
-        objective_power = np.int(
-            self.openslide_wsi.properties[openslide.PROPERTY_NAME_OBJECTIVE_POWER]
-        )
+        props = self.openslide_wsi.properties
+        if openslide.PROPERTY_NAME_OBJECTIVE_POWER in props:
+            objective_power = int(props[openslide.PROPERTY_NAME_OBJECTIVE_POWER])
+        else:
+            objective_power = None
 
         slide_dimensions = self.openslide_wsi.level_dimensions[0]
         level_count = self.openslide_wsi.level_count
         level_dimensions = self.openslide_wsi.level_dimensions
         level_downsamples = self.openslide_wsi.level_downsamples
-        vendor = self.openslide_wsi.properties[openslide.PROPERTY_NAME_VENDOR]
-        mpp_x = self.openslide_wsi.properties[openslide.PROPERTY_NAME_MPP_X]
-        mpp_y = self.openslide_wsi.properties[openslide.PROPERTY_NAME_MPP_Y]
-        mpp = [mpp_x, mpp_y]
+        vendor = props.get(openslide.PROPERTY_NAME_VENDOR)
+
+        # Find microns per pixel (mpp)
+        # Initialise to None (value if cannot be determined)
+        mpp = None
+        # Check OpenSlide for mpp metadata first
+        try:
+            mpp_x = props[openslide.PROPERTY_NAME_MPP_X]
+            mpp_y = props[openslide.PROPERTY_NAME_MPP_Y]
+            mpp = [mpp_x, mpp_y]
+        # Fallback to TIFF resolution units and convert to mpp
```

Comment:
Add a warning message saying the mpp retrieved from tiff metadat.

Reply:
Just added in the latest commit